src/select/autosize-input.js

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import React from 'react';
+import {StyledInput, StyledInputSizer} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+import type {AutosizeInputPropsT, AutosizeInputStateT} from './types';
+
+export default class AutosizeInput extends React.Component<
+  AutosizeInputPropsT,
+  AutosizeInputStateT,
+> {
+  mounted: boolean;
+  input: ?HTMLInputElement;
+  sizer: ?HTMLElement;
+
+  static defaultProps = {
+    inputRef: undefined,
+    value: '',
+  };
+  state = {
+    inputWidth: 5,
+  };
+  componentDidMount() {
+    this.mounted = true;
+    this.updateInputWidth();
+  }
+  componentDidUpdate(
+    prevProps: AutosizeInputPropsT,
+    prevState: AutosizeInputStateT,
+  ) {
+    this.updateInputWidth();
+  }
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+  inputRef = (el: ?HTMLInputElement) => {
+    this.input = el;
+    if (typeof this.props.inputRef === 'function') {
+      this.props.inputRef(el);
+    }
+  };
+  sizerRef = (el: ?HTMLElement) => {
+    this.sizer = el;
+  };
+  updateInputWidth() {
+    if (
+      !this.mounted ||
+      !this.sizer ||
+      typeof this.sizer.scrollWidth === 'undefined'
+    ) {
+      return;
+    }
+    const newInputWidth = this.sizer.scrollWidth + 2;
```

Comment:
what is `2` stands for? Just to invoke update?

Reply:
No, it's just some extra little space around input text.